### PR TITLE
Check open files for modifications

### PR DESCRIPTION
**Pull request by achesak**
Fixed some issues with the previous implementation related to saving, as well as a couple other minor improvements.

### CODE REVIEW

File: aporia.nim [hunk 1]
```diff
@@ -125,11 +125,46 @@ proc updateTabUI(t: Tab) =
   else:
     t.label.setText(name)
   setTabTooltip(t)
+ 
+proc checkFileUpdate(pageNum : int): int =
```
Comment (dom96) [inline aporia.nim]: No spaces before colon.

File: aporia.nim [hunk 2]
```diff
@@ -125,11 +125,46 @@ proc updateTabUI(t: Tab) =
   else:
     t.label.setText(name)
   setTabTooltip(t)
+ 
+proc checkFileUpdate(pageNum : int): int = 
+  ## Checks if the current tab open has been updated outside of Aporia.
```
Comment (dom96) [inline aporia.nim]: *if the currently opened tab

File: aporia.nim [hunk 3]
```diff
@@ -125,11 +125,46 @@ proc updateTabUI(t: Tab) =
...
+  ## Returns ``1`` if the file has been updated, ``0`` if not, and ``-1``
```
Comment (dom96) [inline aporia.nim]: I'm not clear on the difference between ``1`` and ``0``, can you clarify?

Comment (dom96) [inline aporia.nim]: Oh. I see. You ask the user whether they want to reopen it in the same proc. Please separate this logic.

File: aporia.nim [hunk 4]
```diff
@@ -125,11 +125,46 @@ proc updateTabUI(t: Tab) =
...
+  ## if there has been no change
+
+  var changedInfo: FileInfo = getFileInfo(win.tabs[pageNum].filename)
+  result = -1
+  if win.tabs[pageNum].lastEdit != changedInfo.lastWriteTime:
+
+    var filename: string = extractFilename(win.tabs[pageNum].filename)
+    var askUpdate = win.w.messageDialogNew(0, MessageWarning, BUTTONS_NONE, nil)
```
Comment (dom96) [inline aporia.nim]: Please use the same logic to show a message as when a file cannot be opened. This may require some refactoring unfortunately so if you don't want to do it then that's fine. Message boxes are in general annoying and I like to avoid them.

Here is where what I am talking about is used: https://github.com/nim-lang/Aporia/blob/master/aporia.nim#L828

File: aporia.nim [hunk 5]
```diff
@@ -1485,6 +1527,21 @@ proc onSwitchTab(notebook: PNotebook, page: PNotebookPage, pageNum: guint,
   # Toggle the "Syntax Highlighting" check menu item based in the new tabs
   # syntax highlighting.
   plCheckUpdate(pageNum)
+  
+  updateMainTitle(pageNum)
```
Comment (dom96) [inline aporia.nim]: Why was this moved?

File: aporia.nim [hunk 6]
```diff
@@ -1485,6 +1527,21 @@ proc onSwitchTab(notebook: PNotebook, page: PNotebookPage, pageNum: guint,
...
+                      user_data: Pgpointer) = 
+  ## After tab switch; required so the tab is visible when the check for
+  ## modification has run.
+    
+  # Check if the file changed outside the program.
+  if win.tabs[pageNum].filename != "":
+    discard checkFileUpdate(pageNum)
+  
+  updateMainTitle(pageNum)
+  updateStatusBar(win.tabs[pageNum].buffer)
```
Comment (dom96) [inline aporia.nim]: So now we're updating the main title and status bar twice per tab switch?

**Comment by dom96** [comment]
Thanks for this, some changes needed though.

File: aporia.nim (hunk 5 above)
Comment (achesak) [inline aporia.nim]: Consistency with the new function mostly. I'll move it back.

File: aporia.nim (hunk 6 above)
Comment (achesak) [inline aporia.nim]: Necessary to have the UI updated properly. Otherwise the title and status display incorrectly when the dialog appears.

**Comment by achesak** [comment]
As a result of changing from a dialog to an infobar, the check in saveTab() had to be removed as it no longer blocked.

File: aporia.nim [hunk 7]
```diff
@@ -125,6 +125,31 @@ proc updateTabUI(t: Tab) =
   else:
     t.label.setText(name)
   setTabTooltip(t)
+ 
+proc checkFileUpdate(pageNum: int) = 
+  ## Checks if the currently opened tab has been updated outside of Aporia.
+
+  var changedInfo: FileInfo = getFileInfo(win.tabs[pageNum].filename)
+  if win.tabs[pageNum].filename != "" and
```
Comment (dom96) [inline aporia.nim]: This check should be above the call to ``getFileInfo``.

**Comment by dom96** [comment]
I promise that's the last thing for this PR. Please also merge master, fix conflicts and I will merge.

**Comment by achesak** [comment]
Fixed.
